### PR TITLE
Igal/MGMT-2145 Assisted-controller must not try to update already installed nodes or nodes in error

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 1]
```diff
@@ -61,9 +61,16 @@ func NewController(log *logrus.Logger, cfg ControllerConfig, ops ops.Ops, ic inv
 
 func (c *controller) WaitAndUpdateNodesStatus() {
 	c.log.Infof("Waiting till all nodes will join and update status to assisted installer")
-	assistedInstallerNodesMap := c.getInventoryNodesMap()
-	for len(assistedInstallerNodesMap) > 0 {
+	for {
 		time.Sleep(GeneralWaitTimeout)
+		assistedInstallerNodesMap, err := c.ic.GetHosts([]string{models.HostStatusDisabled,
```
Comment (eranco74) [inline src/assisted_installer_controller/assisted_installer_controller.go:66]: This is very misleading.
Consider declaring a var (Ignore/filterStatuses) with this statuses so it will be clear that you don't want these hosts.
Also why did you move into the for loop?

**Review by eranco74 — COMMENTED** [review]
Looks good just a nit

File: src/assisted_installer_controller/assisted_installer_controller.go (hunk 1 above)
Comment (tsorya) [reply src/assisted_installer_controller/assisted_installer_controller.go:66]: cause worker can get error while we are running already. Looks like better to have updated status all the time

**Review by eranco74 — COMMENTED** [review]
/lgtm
